src/contexts/product-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useReducer } from "react";
+import { productsReducer } from "../components/reducers/product-reducer";
+import { products } from "../backend/db/products";
```

Comment:
This is not the preferred way of receiving  the products from the backend. So you need to use the axios dependency or fetch methods to call the APIs provided by backend and then from the response received of these APIs you need to extract the appropriate Products and categories from it